tests/study_tests/test_study.py

```diff
@@ -1437,6 +1437,11 @@ def test_tell_invalid() -> None:
     with pytest.raises(ValueError):
         study.tell(study.ask().number + 1, 1.0)
 
+    # Waiting trial cannot be told.
+    with pytest.raises(ValueError):
+        study.enqueue_trial({})
+        study.tell(len(study.trials) - 1, 1.0)
```

Comment:
A nit, but how about specifying trial number explicitly?
```suggestion
        study.tell(study.trials[-1].number, 1.0)
```